app/derivative_services/image_derivative_service.rb

```diff
@@ -154,10 +156,10 @@ def cleanup_derivative_metadata(derivatives:)
 
     # Updates error message property on the original file.
     def update_error_message(message:)
-      target_file.error_message = [message]
+      resource = change_set_persister.query_service.find_by(id: self.resource.id)
+      target = resource.file_metadata.find { |x| x.id == target_file.id }
+      target.error_message = [message]
       updated_change_set = ChangeSet.for(resource)
-      change_set_persister.buffer_into_index do |buffered_persister|
-        buffered_persister.save(change_set: updated_change_set)
-      end
+      change_set_persister.save(change_set: updated_change_set)
```

Comment:
Sometimes you changed this to pull it out of a `buffer_into_index` and sometimes not. Can you say a little something about each case?

I sort of wonder whether we should pull this method up to a superclass but it appears there isn't one!

Reply:
There was a looping error at one point, but to be honest I think it was just whatever I copied/pasted from. 😂 

I thought about superclass too, but it looked like the "target" was different in different derivative services.

Reply:
I can put 'em back in `buffer_into_index` so there's less of a diff.